daemon/appinfo.c

```diff
@@ -902,6 +934,39 @@ appinfo_parse_desktop(appinfo_t *self)
             appinfo_set_application_name(self, tmp),
             g_free(tmp);
 
+        tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_EXEC_DBUS, 0);
+        if( tmp ) {
+            /* As in libcontentaction, add invoker to the command line if it doesn't exist already. */
+            if( g_strstr_len(tmp, -1, "invoker") != tmp &&
+                    g_strstr_len(tmp, -1, "/usr/bin/invoker") != tmp) {
+                gchar *booster = keyfile_get_string(ini, DESKTOP_SECTION,
+                        NEMO_KEY_APPLICATION_TYPE, NULL);
+                if( booster == NULL || g_strcmp0(booster, "no-invoker") == 0 ) {
+                    /* Default booster type is "generic". This can be overridden via
+                      "X-Nemo-Application-Type=<boostertype>".
+                      "no-invoker" is synonymous to "generic" */
+                    g_free(booster);
+                    booster = g_strdup("generic");
+                }
+
+                gchar *single_instance = keyfile_get_string(ini, DESKTOP_SECTION,
+                        NEMO_KEY_SINGLE_INSTANCE, NULL);
+
+                gchar *exec = tmp;
+                tmp = g_strdup_printf("/usr/bin/invoker --type=%s --id=%s %s %s",
+                                      booster,
+                                      appinfo_id(self),
+                                      g_strcmp0(single_instance, "no") ? "--single-instance" : "",
```

Comment:
If the excess spaces in the invoker cmd line does not matter this is ok and clear/clean.

Otherwise
```
tmp = g_strdup_printf("/usr/bin/invoker --type=%s --id=%s %s%s",
...
g_strcmp0(single_instance, "no") ? "--single-instance " : " ",
```
?

Reply:
Done.